### ISSUE TITLE
Crash when hovering generic type

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1544]
Using the code below, the IDE crashes when you hover over `PropertyType` in the `public Program.PropertyType<Sprite, TProperty>.Result prop;` line.
```cs
public class PropertyType<TTarget, TProperty>
	where TProperty : const String
{
	public typealias Result = comptype(GetPropertyType(TProperty));

	[Comptime]
	public static Type GetPropertyType(String prop)
	{
		if (typeof(TTarget).GetField(prop) case .Ok(let field))
			return field.FieldType;
		if (typeof(TTarget).GetMethod(scope $"get__{prop}") case .Ok(let method))
			return method.ReturnType;
		if (typeof(TTarget).IsGenericParam)
			return typeof(var);
		Runtime.FatalError(scope $"No field or property named {prop}.");
	}
}

struct TestCrash<TProperty>
	where TProperty : const String
{
	public Program.PropertyType<Sprite, TProperty>.Result prop;
}
```
Tested with 5083f5d64eb98fea1ccbb6ec1b63bc3639e6e6b6

**Comment by bfiete** [comment]
Hm. I even tried this with the last nightly and couldn't seem to reproduce it.

Can you attach a crash dump from the nightly build?

**Comment by disarray2077** [comment]
Here it is: [crash1.zip](https://github.com/beefytech/Beef/files/8577589/crash1.zip) (nightly 1abccded)

**Comment by bfiete** [comment]
Fixed at 5f0fd070add24e3b43924fc9816b410e16407f73
